### DISCUSSION TITLE
<Code> doesn't support passing a meta string to Shiki

### DISCUSSION THREAD

**Discussion by jcayzac** [withastro/roadmap#991] [Proposal]
### Astro Info

```block
Astro                    v4.12.2
Node                     v20.15.1
System                   macOS (arm64)
Package Manager          unknown
Output                   static
Adapter                  none
Integrations             astro-icon
                         @astrojs/preact
                         @astrojs/mdx
                         @astrojs/sitemap
```

### Describe the Bug

`<Code>` doesn't support passing a meta string to Shiki's highlighter, breaking the functionality of multiple shiki transformers.

### What's the expected result?

`<Code>` should support passing a meta string to Shiki's highlighter and not break the functionality of multiple shiki transformers.

### Link to Minimal Reproducible Example

https://astro.build/

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by jcayzac** [comment]
I'll send a PR.

**Comment by Princesseuh** [comment]
Thank you for the PR, I'll just move this issue to a roadmap discussion, since it's more of a feature request (and it'll be closed anyway in the next minor, thanks to your PR)

**Comment by Xkonti** [comment]
@Princesseuh Wasn't this functionality released in 4.14? https://astro.build/blog/astro-4140/#support-for-metastrings-in-the-code-component
